neuIPs/neuIP-3.md

```diff
@@ -0,0 +1,53 @@
+---
+eip: 3
+title: Music NFT Release Event
+description: A minimal primitive to emit URI events for social media.
+author: Kevin Neaton (@neatonk), Tim Daubenschütz (@TimDaub)
+discussions-to: https://github.com/neume-network/neuIPs/issues/4
+status: Draft
+created: 2022-09-16
+---
+
+## Abstract
+
+We introduce a minimal on-chain primitive to emit pointers to music NFTs using Ethereum's event log infrastructure.
+
+## Motivation
+
+The modern social network's base layer is distributing content behind Universal Resource Identifiers as defined by RFC 3986. Any web resource can be identified by an URI or a URL and so upgrading Ethereum with a standard way of distributing content is useful.
+
+## Specification
+
+```solidity
+contract Feed {
+  event NewTrack(address indexed curator, string indexed uri);
+  function release(string calldata uri) external {
+    emit NewTrack(msg.sender, uri);
+  }
+}
+```
+
+- We strongly recommend all implementers to expose an additional [EIP-165](./eip-165.md) interface.
+- The input `string calldata uri` of `function release(...)` must be an URI as defined by RFC 3986.
+
+### NFT identification
+
+An NFT is identified by its compliance to [EIP-721](./eip-721.md), its address
+and its `uint256 tokenId` and can be represented as an URI:
+
+```
+nft://erc721:0x06012c8cf97BEaD5deAe237070F9587f8E7A266d/771769
```

Comment:
Is there an EIP or RFC for the `nft` uri scheme being used here?

Reply:
not really, but there are two that we could use if we wanted to: 

- https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-19.md
- https://eips.ethereum.org/EIPS/eip-4804

But to me no URI scheme is currently final and so we might as well just:

- come up with our own differentiated scheme
- decide to ignore it for the sake of this standard
- ???

Reply:
I'm in favor of ignoring the uri scheme for the sake of this standard and making it more generic. I like how the uri scheme can be used when processing events to quickly filter for the type of thing you're interested in. E.g. NFTs.

Reply:
ok, instead of proposing a specific uri scheme, can u please edit the PR to remove the section in question.

still, do we have any normative requirements in regard to the URI users are posting? E.g. should the URI always link to an NFT?

Reply:
I think not. But a non-normative section with examples would help to illustrate the indented our use case.

Reply:
ok sounds good

Reply:
[TANGENT] That said I think there could be value in unique [event-name, uri-scheme] pairings as a way to partition the event stream. This makes sense because the event processor can verify that the right uri scheme is being used... in fact adding the scheme as an indexed event topic would be even better for filtering.

Reply:
EIP-3722 has a `string indexed tag` would that help for partitioning?

Reply:
lets emit `Track(address contract, uint256 tokenId)`

Reply:
For editions it may be best to omit the tokenId, even though neume doesn't really support that yet. E.g. `Track(address contract, uint256 tokenId, bool hasTokenId)`

Reply:
> For editions it may be best to omit the tokenId, even though neume doesn't really support that yet. E.g. `Track(address contract, uint256 tokenId, bool hasTokenId)`

I have clarified this with:

> Although a song might be represented by many tokens, for all, the event NewSong must only be emitted once on the network and preferrably with its lowest locatable uint256 tokenId.

So for editions, we've be using the lowest tokenId. OK?